components/wso2is7.key.manager/src/main/java/org/wso2/is7/client/WSO2IS7KeyManager.java

```diff
@@ -1578,7 +1595,19 @@ private List<String> getRoles(Scope scope) {
         }
         return Collections.emptyList();
     }
-
+    /**
+     * Retrieves the WSO2 IS7 role name based on the provided role name.
+     * If role creation is disabled, the original role name is returned.
+     * When role creation is enabled, the method applies specific naming conventions:
+     * - Removes the "Internal/" prefix if present.
+     * - Throws an exception if the role starts with "Application/".
+     * - Returns the role name as-is if it starts with "system_primary_".
```

Comment:
I guess this is not applicable anymore?
`- Returns the role name as-is if it starts with "system_primary_".`